mmv1/templates/tgc_next/cai2hcl/resource_converter.go.tmpl

```diff
@@ -109,11 +109,9 @@ func (c *{{ $.ResourceName -}}Cai2hclConverter) convertResourceData(asset caiass
 	}
 {{ end}}
 
-{{ range $param := $.UserParameters }}
-{{/* Attempt to parse all parameters from asset name. */}}
-	hclData["{{ underscore $param.Name }}"] = utils.ParseFieldValue(asset.Name, "{{ underscore (plural $param.Name) }}")
-{{ end }}
-{{- if $.HasProject -}}
+{{/* Attempt to parse all self-link parameters from asset name. */}}
+	utils.ParseUrlParamValuesFromAssetName(asset.Name, "{{ $.SelfLinkUri }}", hclData)
+{{ if $.HasProject -}}
 	hclData["project"] = utils.ParseFieldValue(asset.Name, "projects")
```

Comment:
```suggestion
```
We could probably delete this since every resource with a project should also have a project in its `SelfLinkUri`

Reply:
We can do it now and handle the edge cases later if any.